driver: wifi: eswifi: Various cleanups

parthitce opened the pull request:
use common log module across all eswifi sources, which allows
to compile with CONFIG_NET_SOCKETS_OFFLOAD.

Also fix k_timeout_t compilation failure when CONFIG_NET_SOCKETS_OFFLOAD
enabled and warnings.

Signed-off-by: Parthiban Nallathambi <parthiban@linumiz.com>

parthitce commented:
```
CONFIG_NET_SOCKETS=y
CONFIG_NET_SOCKETS_OFFLOAD=y
```

These 2 Kconfig entry is not enabled by default for eswifi under wifi sample. So these compilation errors wasn't caught in previously. Should we define these under `samples/net/wifi/boards/disco_l475_iot1.conf` by default?

jukkar reviewed (changes requested):
As there are two distinct changes in this PR, please separate the log changes from k_timeout_t one, so two commits. Or perhaps three/four as you have mysterious cast and null check in there too.

jukkar commented:
> These 2 Kconfig entry is not enabled by default for eswifi under wifi sample. So these compilation errors wasn't caught in previously. Should we define these under `samples/net/wifi/boards/disco_l475_iot1.conf` by default?

Wouldn't these be needed for all the boards? Why not put them to prj.conf file?

parthitce commented:
> As there are two distinct changes in this PR, please separate the log changes from k_timeout_t one, so two commits. Or perhaps three/four as you have mysterious cast and null check in there too.

Sure, will split it. Thanks

parthitce commented:
> > These 2 Kconfig entry is not enabled by default for eswifi under wifi sample. So these compilation errors wasn't caught in previously. Should we define these under `samples/net/wifi/boards/disco_l475_iot1.conf` by default?
> 
> Wouldn't these be needed for all the boards? Why not put them to prj.conf file?

Not sure here. ESP and winc1500 can't use this option if I am not wrong.

jukkar commented:
> > Wouldn't these be needed for all the boards? Why not put them to prj.conf file?
> 
> Not sure here. ESP and winc1500 can't use this option if I am not wrong.

Ok, in that case it is probably best to put the options to eswifi driver Kconfig so that they will be always selected if they are always needed for this driver.

parthitce committed:
driver: wifi: eswifi: use common log module

use common log module across all eswifi sources, which allows
to compile with CONFIG_NET_SOCKETS_OFFLOAD.

Signed-off-by: Parthiban Nallathambi <parthiban@linumiz.com>

parthitce committed:
drivers: wifi: eswifi: Convert raw timeout values to use K_MSEC()

Use K_MSEC instead of raw timeout value.

Signed-off-by: Parthiban Nallathambi <parthiban@linumiz.com>

parthitce committed:
drivers: wifi: eswifi: fix compilation warnings

Fix compilation warnings when CONFIG_NET_SOCKETS_OFFLOAD enabled

warning: passing argument 5 of '__eswifi_socket_new' makes pointer
 from integer without a cast [-Wint-conversion]
  395 |  idx = __eswifi_socket_new(eswifi, family, type, proto, 0x50CE);
      |                                                         ^~~~~~
      |                                                         |
      |                                                         int
eswifi.h:137:21: note: expected 'void *' but argument is of type 'int'
  137 |    int proto, void *context);
      |               ~~~~~~^~~~~~~
In function 'eswifi_socket_bind':
warning: comparison between pointer and integer
  461 |  if ((addrlen == NULL) || (addr == NULL) ||
      |

Signed-off-by: Parthiban Nallathambi <parthiban@linumiz.com>

parthitce committed:
drivers: wifi: eswifi: enable NET_SOCKETS_OFFLOAD by default

Enable NET_SOCKETS_OFFLOAD is default option for eswifi.

Signed-off-by: Parthiban Nallathambi <parthiban@linumiz.com>

parthitce commented:
@jukkar Thanks for the review. Commit are split and NET_SOCKETS_OFFLOAD enabled by default.

jukkar commented on drivers/wifi/eswifi/eswifi_socket_offload.c:
```diff
@@ -392,7 +392,7 @@ static int eswifi_socket_open(int family, int type, int proto)
 
 	/* Assign dummy context SOCkEt(50CE) */
-	idx = __eswifi_socket_new(eswifi, family, type, proto, 0x50CE);
+	idx = __eswifi_socket_new(eswifi, family, type, proto, (void *)0x50CE);
```
What is this magic value? Could you use here `INT_TO_POINTER()` instead?

parthitce replied:
@jukkar socket offloading maintains the zephyr's net_context struct and translates into AT communication to use the TCP/IP stack on chip. ISM43362-M3G-L44-E specifically can handle maximum 4 simultaneous socket connection and it's pre-allocated. 0x50CE initialized to net_context when new socket is created/opened and when more than 4 connections needed, open/socket needs to wait until it gets one free slot.

There is no special meaning to this 0x50CE magic value apart from it, so can be placed to macro and use INT_TO_POINTER. I will change it now.

pfalcon commented on drivers/wifi/eswifi/eswifi_socket_offload.c:
```diff
@@ -30,6 +30,7 @@ LOG_MODULE_DECLARE(LOG_MODULE_NAME);
  */
 #define SD_TO_OBJ(sd) ((void *)(sd + 1))
 #define OBJ_TO_SD(obj) (((int)obj) - 1)
+#define ESWIFI_INIT_CONTEXT	INT_TO_POINTER(0x50CE)
```
This only deteriorates the situation, at least previously there was a comment alongside this magic number: `/* Assign dummy context SOCkEt(50CE) */` (character case problems?), now it's the magic number all on its own.

parthitce replied:
@pfalcon What's the best fix here? Makes sense to move the comment to top as well?

pfalcon replied:
> What's the best fix here? Makes sense to move the comment to top as well?

Yes, that's what I'd do.

parthitce replied:
@pfalcon Thanks, moved it.

pfalcon reviewed (commented):
@parthitce: The title of this PR, "driver: wifi: use common log module" doesn't correspond to its content: it doesn't patch all wifi drivers to use common log module, and for the one driver it patches (eswifi), it makes changes beyond "use common log module".

Is this intentional? If so, please update the title accordingly. Otherwise, it looks like unrelated commits got into this PR, which need to undergo separate reviews (as some of them are questionable/may need further cleanup).

Thanks.

parthitce commented:
> @parthitce: The title of this PR, "driver: wifi: use common log module" doesn't correspond to its content: it doesn't patch all wifi drivers to use common log module, and for the one driver it patches (eswifi), it makes changes beyond "use common log module".
> 
> Is this intentional? If so, please update the title accordingly. Otherwise, it looks like unrelated commits got into this PR, which need to undergo separate reviews (as some of them are questionable/may need further cleanup).
> 
> Thanks.

No, it wasn't intentional. Initial revision clubs all the changes in one single commit and later split into multiple units based on @jukkar comment and I missed updating the title. Now updated, but still as the changes are discrete title may not cover all the things the PR does here. Thanks

pfalcon commented:
> Now updated, but still as the changes are discrete title may not cover all the things the PR does here. 

Thanks for updating the PR title. Something like "driver: wifi: eswifi: Various cleanups" would work too, to set reviewers' expectations right.

parthitce commented:
> > Now updated, but still as the changes are discrete title may not cover all the things the PR does here.
> 
> Thanks for updating the PR title. Something like "driver: wifi: eswifi: Various cleanups" would work too, to set reviewers' expectations right.

@pfalcon Thanks for the suggestions, updated the title.

loicpoulain commented:
@parthitce have you tested these changes with wifi/net shell?

parthitce commented:
> @parthitce have you tested these changes with wifi/net shell?

@loicpoulain Was working on older master. Rebasing to latest failed to work now. git bisect shows 62580748e536d5ff14ecc290e114c034f1ccccfd (from https://github.com/zephyrproject-rtos/zephyr/pull/26349). I will check this. DNM please!

pfalcon commented:
@parthitce: Thanks for the updates.

@loicpoulain: I assume that you'll keep an eye on this PR and approve once it's confirmed to work. (I just try to clear my review backlog before going on vacation.)

parthitce commented:
@pfalcon @loicpoulain https://github.com/zephyrproject-rtos/zephyr/pull/27041 fixes the issue.
